spec/latest/index.bs

```diff
@@ -224,6 +225,246 @@ Pairs must not be duplicated.
+array at position `p` in a dense level of rank `r` is an array whose slice
+
+`A[i_0, ..., i_(r - 1), :, ..., :]`
+
+is described by the row-major position
+
+`q = (((((p * I_0) + i_0) * I_1) + i_1) * I_2 + i_2) * ... + i_(r - 1)`
+
+of the sublevel.
+
+#### Sparse #### {#sparse_level}
+
+If the level is an sparse level, the `subformat` key must be present.  The
+`rank` key must be present, and set to an integer `r` greater than or equal to
+`1`.  The sparse level represents zero or more `r`-dimensional sparse arrays
+whose non-implicit elements are themselves arrays specified by `subformat`. For
+example, a sparse level of rank 1 represents a collection of sparse vectors of
+subarrays.
+
+Assume that this level represents `n`-dimensional subarrays and the root array
+is `N`-dimensional.  The sparse level implies the following binary arrays are
+present:
+
+: pointers_(N - n - 1)
+:: Array of size `number_of_positions + 1` whose 1st element is equal to `0` and whose `p + 1`th element is equal to the sum of `pointers_N[p]` and the number of explicitly represented slices in the `p`th position.
+
+: indices_(N - n), ..., indices(N - n + r - 1)
+:: There are `r` such arrays. When `A[i_0, ..., i_(r - 1), :, ..., :]` is explicitly represented by the subarray in position `q`, `indices_(N-n+s)[q] = i_s`. The arrays must be ordered such that the tuples `(indices_(N-n)[q], ..., indices_(N-n+r-1)[q])` are unique and appear in lexicographic order for all `q` in each range `pos[p] <= q < pos[p + 1]`. This array must contain no other elements.
+
+Special note: If the sparse level is the root level, the `pointers` array should
+be ommitted, as its first value will be `0` and its last value will be the
+length of any of the `indices` arrays in this level.
+
+#### ISO #### {#iso_level}
+
+When all values of a sparse array are the same identical value, a special level
+is provided to compress the value array to a single value rather than
+duplicating the same number unnecessarily. Iso levels represents zero or more
+scalars of all the same value. It implies that the following binary arrays are
+present in the tensor:
+
+: values
+:: Array of size `1` whose element holds the value of all explicit entries in the sparse tensor.
+
+Note: Structure-only matrices (allowed in matrix market format) can be stored
+      using this level with a value of 1. This adds only a small amount of
+      overhead while describing essentially the same matrix.
+
+### Equivalent Formats ### {#equivalent_formats}
+
+The following formats are equivalent
+
+#### VEC #### {#vec_format_equiv}
+
+Vector format is equivalent to
+
+```json
+{
+  "format": {
+    "level": "dense",
+    "rank": "1",
+    "subformat": {
+      "level": "element",
+    }
+  },
+}
+```
+
+#### CSR #### {#csr_format_equiv}
+
+Compressed-Sparse Row format is equivalent to
+
+Vector format is equivalent to
+
+```json
+{
+  "format": {
+    "level": "dense",
+    "rank": "1",
```

Comment:
Small nit: the current spec uses integers for integers instead of strings. For example, `1` instead of `"1"`. We should probably keep to this convention for integers in metadata.

Reply:
oh oops! I think that was just a mistake in the spec, the implementation was using ints. I'll fix it up. Thanks for reading!